lua/lsp/tsserver-ls.lua

```diff
@@ -1,15 +1,28 @@
 vim.cmd "let proj = FindRootDirectory()"
 local root_dir = vim.api.nvim_get_var "proj"
 
--- use the global prettier if you didn't find the local one
-local prettier_instance = root_dir .. "/node_modules/.bin/prettier"
-if vim.fn.executable(prettier_instance) ~= 1 then
-  prettier_instance = O.lang.tsserver.formatter.exe
+local formatter_exe = O.lang.tsserver.formatter.exe
+
+local get_prettier_module_name = function()
+  -- get local module name
+  if formatter_exe == "prettierd" then
+    return "@fsouza/prettierd"
+  end
+  return formatter_exe
+end
+
+local get_prettier_instance = function()
+  -- prioritize local instance over global
+  local local_instance = root_dir .. "/node_modules/.bin/" .. get_prettier_module_name()
+  if vim.fn.executable(local_instance) == 1 then
+    return local_instance
+  end
+  return formatter_exe
 end
 
 O.formatters.filetype["javascriptreact"] = {
   function()
-    local args = { "--stdin-filepath", vim.fn.fnameescape(vim.api.nvim_buf_get_name(0)) }
```

Comment:
are you sure `prettier` doesn't need this argument?
```
--stdin-filepath <path>  Path to the file to pretend that stdin comes from.
```

Reply:
Good question. I had to remove `--stdin-filepath` for `prettierd` (otherwise it didn't work)
I also tried with normal prettier to make sure it still work and everything seems ok.
Should we put this argument condtionally if using `prettier`? idk

Reply:
I don't know either 🤦 , somebody else can help probably 🥺

Reply:
According to the documentation https://github.com/fsouza/prettierd#editor-integration we only need to give `vim.api.nvim_buf_get_name(0)` to args !

```lua
-- prettierd
       function()
          return {
            exe = "prettierd",
            args = {vim.api.nvim_buf_get_name(0)},
            stdin = true
          }
        end
```

Reply:
Thanks for the information - does `prettier` still need the argument tho? If yes, we should put the argument conditonally then. Everything works as expected on `prettier` and `prettierd` without the argument on my side, but I don't want to break anything 🙃

Reply:
@williamgoulois can you test this with the `prettier` as well, just wanna make sure that it doesn't need the `--stdin-filepath`, bear in mind that instead of directly opening the file, you need to open it through telescope file picker or sth to be sure

Reply:
Just tested with prettier local and global this does work !